### ISSUE TITLE
switching from view to table yields late error

### ISSUE THREAD

**Issue by HarlanH** [fishtown-analytics/dbt#499]
Currently, if you develop a model as a view, then switch to materializing into a table, you get an error that looks like this:

```
2017-08-02 15:30:50,418: On fct_job_listing_email_activity: drop table if exists "dbt_harlan"."fct_job_listing_email_activity" cascade
2017-08-02 15:30:50,551: Postgres error: "fct_job_listing_email_activity" is not a table
HINT:  Use DROP VIEW to remove a view.
```

That's correct, and clear enough, but it happened _after_ the temp table was built, which in this case takes quite a while. My expectation was that either the system would be smart enough to drop a view when building a table (no error), or that it would detect that the target was a view before going ahead and building the temp table (early error). Could one of those fixes be added?

**Comment by drewbanin** [comment]
Hey @HarlanH -- thanks for creating this issue. This is a regression in v0.8.3, dbt should (and used to) issue the correct drop statement. It looks like `table` is hardcoded [here](https://github.com/fishtown-analytics/dbt/blob/master/dbt/materializers.py#L88)

This works differently in development now, but I'll confirm that the new code handles this appropriately. Thanks!

**Comment by drewbanin** [comment]
@HarlanH I just confirmed that this is working correctly in `development`. This fix will go out in our next release, `0.9.0`!
